src/Adapter/Product/Image/ProductImagePathFactory.php

```diff
@@ -28,46 +28,87 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Product\Image;
 
-use Image;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\ValueObject\ImageId;
 
 class ProductImagePathFactory
 {
+    public const IMAGE_TYPE_SMALL_DEFAULT = 'small_default';
+    public const IMAGE_TYPE_MEDIUM_DEFAULT = 'medium_default';
+    public const IMAGE_TYPE_LARGE_DEFAULT = 'large_default';
+    public const IMAGE_TYPE_HOME_DEFAULT = 'home_default';
+    public const IMAGE_TYPE_CART_DEFAULT = 'cart_default';
+
     /**
-     * @var bool
+     * @var string
      */
-    private $isLegacyImageMode;
+    private $temporaryImgDir;
 
     /**
      * @var string
      */
-    private $temporaryImgDir;
+    private $pathToBaseDir;
+
+    /**
+     * @var string
+     */
+    private $contextLangIsoCode;
 
     /**
-     * @param bool $isLegacyImageMode
+     * @param string $pathToBaseDir
      * @param string $temporaryImgDir
+     * @param string $contextLangIsoCode
      */
     public function __construct(
-        bool $isLegacyImageMode,
-        string $temporaryImgDir
+        string $pathToBaseDir,
+        string $temporaryImgDir,
+        string $contextLangIsoCode
     ) {
-        $this->isLegacyImageMode = $isLegacyImageMode;
-        $this->temporaryImgDir = $temporaryImgDir;
+        // make sure one trailing slash is always there
+        $this->temporaryImgDir = rtrim($temporaryImgDir, '/') . '/';
+        $this->pathToBaseDir = rtrim($pathToBaseDir, '/') . '/';
+        $this->contextLangIsoCode = $contextLangIsoCode;
+    }
+
+    /**
+     * @param ImageId $imageId
+     * @param string $extension
+     *
+     * @return string
+     */
+    public function getPath(ImageId $imageId, string $extension = 'jpg'): string
+    {
+        $path = $this->getBaseImagePathWithoutExtension($imageId);
+
+        return sprintf('%s.%s', $path, $extension);
     }
 
     /**
-     * @param Image $image
+     * @param ImageId $imageId
+     * @param string $type
+     * @param string $extension
      *
      * @return string
      */
-    public function getBasePath(Image $image): string
+    public function getPathByType(ImageId $imageId, string $type, string $extension = 'jpg'): string
```

Comment:
Should we use a const for jpg or it's overkill?

Reply:
I can add one